fennel/client/client.py

```diff
@@ -739,3 +717,32 @@ def _post(
         )
         check_response(response)
         return response
+
+
+def _s3_connector_json(s3: S3Connector) -> Dict[str, Any]:
```

Comment:
No action required, but sharing some raw thoughts (would be nice to add a TODO here and at least keep it open with the context written below) - 

I think we should make this proto encoded into bytes and on the server side decode the bytes back into proto -> struct.

The idea is, this is extremely brittle (i.e. easy to make this code backward breaking or forward breaking). Say we introduce a new field on the server side, requests from older client would break. older server side code would break for newer clients in JSON parsing (because a field would just not be present). This is probably fine for HTTP Requests in general, but since we are using a datastructure (S3Table), which is not constrainted/limited to Extract historical, this could break in the future easily because of them being different code paths or general workflows (so even things like e2e testing may not capture it - since we don't have any extract historical support in client - backend integration tests).

Reply:
I definitely appreciate that concern but if we go to protos then don't we lose the ability to make direct REST calls from cURL / postman?